provisioner/data.js

```diff
@@ -529,6 +521,10 @@ WorkerType.prototype.determineSpotBids = function(regions, pricing, runningCapac
   
   var change = this.determineCapacityChange(runningCapacity, pendingCapacity, pending);
 
+  // TODO: If change is negative we should cancel spot requests... This should
+  // go through the aws-manager. IMO this method belongs in the provision.js
+  // file...
```

Comment:
If we do this... then we're compat with existing provisioner when `scalingRatio = 0` and should roll this out...

I suggest you move `determineSpotBids` into `provsion.js` and add the support for cancelling spot requests.

We may have iterations where we alternate between requesting and canceling spot requests.. Which IMO is fine... We can optimize by prefering nodes that has capacity that isn't above the change we want. Ie. reducing utility by the capacity to desiredChange ratio. But that's not important for now...

Get cancel spot requests done right... this is something we do while running. if `change < 0`.
Then roll this out, it looks really good in general!